### PR TITLE
Check MCG.Reconcile strategy before deleting Noobaa resources

### PR DESCRIPTION
**Pull request by nbalacha**
Delete Noobaa resources only they are being managed
by the OCS operator.

Signed-off-by: N Balachandran <nibalach@redhat.com>

### CODE REVIEW

**Comment by nbalacha** [comment]
Should the same checks be done for the other Managed resources as well?

**Comment by jarrpa** [comment]
I think this makes sense thus far.

> Should the same checks be done for the other Managed resources as well?

Now that you mention it, that doesn't sound like a bad idea. @raghavendra-talur or @iamniting do either of you have any thoughts on that?

File: vendor/modules.txt [hunk 1]
```diff
@@ -153,6 +153,7 @@ github.com/onsi/gomega/matchers/support/goraph/node
 github.com/onsi/gomega/matchers/support/goraph/util
 github.com/onsi/gomega/types
 # github.com/openshift/api v3.9.1-0.20190924102528-32369d4db2ad+incompatible => github.com/openshift/api v0.0.0-20190924102528-32369d4db2ad
+github.com/openshift/api/config/v1
```
Comment (iamniting) [inline vendor/modules.txt:156]: It looks like this Change is not part of this PR

**Review by iamniting — CHANGES_REQUESTED** [review]
> I think this makes sense thus far.
> 
> > Should the same checks be done for the other Managed resources as well?
> 
> Now that you mention it, that doesn't sound like a bad idea. @raghavendra-talur or @iamniting do either of you have any thoughts on that?

Make sense to me

**Comment by iamniting** [comment]
> Delete Noobaa resources only they are being managed
> by the OCS operator.

@nbalacha I think 'if' is missing here.

Can you pls prefix "StorageCluster:" as we are following that pattern for all our delete or uninstall commits.

[1]. https://github.com/openshift/ocs-operator/blame/fac013610cf1bf1f0f50ae9c7f17d2163e004055/pkg/controller/storagecluster/uninstall_reconciler.go

File: deploy/deploy-with-olm.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ spec:
...
     mediatype: image/svg+xml
-  image: quay.io/ocs-dev/ocs-registry:latest
+  image: quay.io/nibalach/ocs-registry:master
```
Comment (iamniting) [inline deploy/deploy-with-olm.yaml:29]: Same as before

**Review by iamniting — CHANGES_REQUESTED** [review]

**Pull request closed without merging**